pkg/ddl/foreign_key.go

```diff
@@ -378,8 +379,8 @@ func isAcceptableForeignKeyColumnChange(newCol, originalCol, relatedCol *model.C
 	return true
 }
 
-func checkTableHasForeignKeyReferred(is infoschema.InfoSchema, schema, tbl string, ignoreTables []ast.Ident, fkCheck bool) *model.ReferredFKInfo {
-	if !fkCheck {
+func checkTableHasForeignKeyReferred(is infoschemactx.MetaOnlyInfoSchema, schema, tbl string, ignoreTables []ast.Ident, fkCheck bool) *model.ReferredFKInfo {
+	if !fkCheck || is == nil {
```

Comment:
why it can be nil now?

Reply:
Because in this PR, the value of returned by `metabuild.Context.GetInfoSchema()` can be nil: https://github.com/pingcap/tidb/pull/56176/files#diff-95d4af02b1eef6f4dc82de49e642785bab3c2d5b60c1de45e3afbf29a6bb2dd8R232-R236

In the previous implementation, it uses a `mock.Context`, which always returns a non-nil infoschema value (but not a right value, it contains some tables for testing...).


However, the assumption that the input value can be nil is not good. I changed the code here: 

https://github.com/pingcap/tidb/pull/56176/commits/9d24559d998628ec49c3179f03e78c508518f0f9

So that, `checkTableHasForeignKeyReferred` still assumpts is is not nil